src/core/searchUtils.js

```diff
@@ -92,6 +92,22 @@ export function loadSearchResultsIfNeeded({ store: { dispatch, getState }, locat
     ...location.query,
     clientApp: state.api.clientApp,
   });
+
+  if (!isLoaded({ state: state.search, page, filters })) {
+    return performSearch({ dispatch, page, api: state.api, auth: state.auth, filters });
+  }
+  return true;
+}
+
+export function loadCategoryResultsIfNeeded({ store: { dispatch, getState }, location, params }) {
+  const page = parsePage(location.query.page);
+  const state = getState();
+  const filters = {
+    addonType: params.addonType,
+    category: params.slug,
+    clientApp: params.application,
+  };
+
   if (!isLoaded({ state: state.search, page, filters })) {
```

Comment:
Good point. Again, it was like that when I got here; the search page currently caches results. When flipping through paginated results it makes going back and forth between pages quick.

I've filed #1404, but I'll remove it from this bit of code because why not and I agree; we can always add it back.